src/shared/copy.c

```diff
@@ -856,7 +856,7 @@ static int fd_copy_fifo(
                      AT_SYMLINK_NOFOLLOW) < 0)
                 r = -errno;
 
-        if (fchmodat(dt, to, st->st_mode & 07777, 0) < 0)
+        if (fchmodat_best(dt, to, st->st_mode & 07777, 0) < 0)
                 r = -errno;
```

Comment:
Please don't use errno here. This is our own wrapper that returns negative errno instead.